### ISSUE TITLE
Promise warnings when plugin imported in support/index.ts

### ISSUE THREAD

**Issue by thibleroy** [bahmutov/cypress-watch-and-reload#60]
**Logs and screenshots**
![](https://github.com/thibleroy/cypress-code-coverage-bug/blob/master/images/promised.PNG?raw=true)
**versions**

- Cypress 4.4.1
- Angular 9.1.3
- code-coverage plugin : 3.5.3
- os : windows 10
- shell : powershell
- Node v12.11.1
- npm v6.11.3

**describe the bug**

When plugin is imported in support/index.ts, a lot of warnings are displayed in the dashboard console.

**link to the repo**
[repo](https://github.com/thibleroy/cypress-code-coverage-bug)

**steps to reproduce**

Import plugin in support/index.ts

**workaround**

- set the environment in cypress.json when working using TDD : 

```
"env": {
    "TDD": true
  }
```

- Only import the plugin when env.TDD is set to `true` : 

```
if (Cypress.env('TDD')) {
    import('cypress-watch-and-reload/support');
}
```

**Comment by jcamden** [comment]
It took me a while to narrow this down to cypress-watch-and-reload. If this is a tough fix, I would vote for adding a warning (about the warnings) to the cypress-watch-and-reload README in order to save people some time in the future.

**Comment by jcamden** [comment]
By the way, if anybody new to Typescript wants to implement @thibleroy's workaround:

```
declare global {
  namespace Cypress {
    interface Chainable<Subject> {
      // custom commands here
    }
    interface env {
      TDD: boolean;
    }
  }
}
```

**Comment by DamienCassou** [comment]
I have the same problem.

**Comment by DamienCassou** [comment]
This should be fixed by PR #83.

**Comment by bahmutov** [comment]
merged https://github.com/bahmutov/cypress-watch-and-reload/pull/83 to fix this
